### ISSUE TITLE
empty_street_name_labels not suited for foreign language (italian)

### ISSUE THREAD

**Issue by paolobenve** [valhalla/odin#399]
The english definition

`"empty_street_name_labels": ["the walkway", "the cycleway", "the mountain bike trail"],`

isn't suited for italian, because a phrase like "Head north on the mountain bike trail" is traduced in italian "procedi verso nord sulla pista per mountain bike": i.e., the determinative article "the" ("il" or "la", masculine or femenine) is in italian joined with the preposition "su" (corresponding to the english "on"): "su + la" = "sulla".

In general, articles join with various prepositions, as shown in http://images.treccani.it/enc/media/share/images/orig/system/galleries/Grammatica/P/preposizioni_tab_p_00072_001.jpg, and this requires that prepositions be joined in the same item where the article is traduced.

**Comment by kevinkreiser** [comment]
yes for some combinations our translations and directions can be somewhat robotic (dont sound like a real human speaking). in german for example we also drop some articles in certain places to avoid solving the issue of varying gender and case. if a general rule can be arrived at in this thread, we do have the ability to implement it in code. essentially we can write code to do the right thing if its more complex than just string replacement. do you think you could formulate a rule for us to implement to handle this preposition merging with an article case? or is it as simple as following the table you linked?

one possible rule/algorithm for this would be:
1. look for `su` preceeding any `empty_street_name_labels`
2. if there was a `su` and the first word in `empty_street_name_labels` is found in that table
3. then replace `su` and the first word by the combination of the two found in the table

**Comment by paolobenve** [comment]
The italian issues we should manage are just the ones summarized in the table, but others issue could appear.

The possible rule/algorithm you write should be applyed with all the propositions affected, and after merging the translated strings, and only for italian.

More generally, I'd define a bidimensional array, with the language as first dimension and the substitutions: "<preposition> <article>" => "<articulated preposition>" as the second one; and after the translation is performed I'd search & replace according to the array, checking for whole words.

This way german and other languages issues could be managed too.
